src/main/scala/com/fulcrumgenomics/fasta/CollectAlternateContigNames.scala

```diff
@@ -178,10 +184,22 @@ class CollectAlternateContigNames
         AssemblyReportColumn.values.foreach { column =>
           attributes += ((column.tag, dict(column.key)))
         }
+        // get the list of aliases (may update attributes if an existing contig is found)
+        val aliases: Seq[String] = existingSequenceDictionary match {
+          case None           => alts
+          case Some(sequenceDictionary) =>
```

Comment:
I don't think this belongs here for a couple of reasons:
1. It's weird to do all this validation when in theory just collecting the list of aliases for a sequence
2. Without more hoop-jumping you can't update a dictionary that doesn't contain the matching set of sequences for the assembly report / sequence-role(s).

I think it would be much cleaner to revert this section and have the main loop just generate a new set of `SequenceMetadata` objects from the assembly report and then do the merging in one single pass down below.